Automattic/newspack-network
# feat(backfill): WP CLI script to backfill reader_registered, donation_new events

### All Submissions:






### Changes proposed in this Pull Request:



Adds a WP CLI script to backfill historical data. 

### How to test the changes in this Pull Request:

1. Switch `newspack-plugin` to `feat/tweaks-for-network-update` branch (https://github.com/Automattic/newspack-plugin/pull/2935)
2. Disable this plugin on both the hub and a node site, in order to build up historical data
3. On both sites (using different email addresses): 
    1. register a user, 
    2. make a donation
    3. make a recurring donation and the cancel the related subscription in WC
5. Activate the `newspack-network` plugin on both sites
6. Verify that the new user registrations and donations are not present in the Hub's event log
7. Run `wp newspack-network data-backfill reader_registered --start=2024-01-01 --end=2024-12-31` on the Hub site, observe the output the dry run and confirm nothing was added to the event log. 
    1. Do the same with `donation_new` as the argument instead of `reader_registered`.
    2. And then with `donation_subscription_cancelled` 
9. Try with `--verbose` flag for more output
10. Run `wp newspack-network data-backfill reader_registered --start=2024-01-01 --end=2024-12-31 --live` on the Hub site and observe the event was added to the event log. 
    1. Do the same with `donation_new` as the argument instead of `reader_registered`.
    2. And then with `donation_subscription_cancelled` 
12. Repeat the last three steps on the Node site, but instead of the event log look as the scheduled events in the Node Settings
13. Run in the live mode _again_ to ensure that duplicate events are not added to the event log or the scheduled events* (in the CLI output, observe it's noted that no events were processed and how many were skipped as duplicates)
14. After the scheduled event is transmitted, observe it was added to the Hub's event log
15. Run the script with `all` as action name argument and observe it has the same effect as running the above two scripts one after another

\* unless the webhook request queue has be cleared already, but the event log should handle deduplication itself anyway

### Other information:
